### PR TITLE
Reliable settings framework

### PR DESCRIPTION
**Pull request by esteldunedain**
Apparently we have problems with settings not being reliably inited before client post init handlers are run.

This creates a `SettingsInitialized` event once every setting has been received from the server, profile,etc. This event is guaranteed to happen **AFTER** all postInits are ready. Hence it's safe to add listener for it on the postInits of the different modules to do many first time inits that require settings.

### CODE REVIEW

File: addons/common/XEH_preInit.sqf [hunk 1]
```diff
@@ -291,9 +291,22 @@ GVAR(syncedEvents) = HASH_CREATE;
 //Debug
 ACE_COUNTERS = [];
 
-// Load settings
+// Wait for server settings to arrive
+GVAR(SettingsInitialized) = false;
+["ServerSettingsReceived", {
```
Comment (Glowbal) [inline addons/common/XEH_preInit.sqf]: This event is never raised?

File: addons/common/XEH_preInit.sqf [hunk 2]
```diff
@@ -291,9 +291,22 @@ GVAR(syncedEvents) = HASH_CREATE;
...
+        call FUNC(loadSettingsLocalizedText);
+    };
+    GVAR(SettingsInitialized) = true;
+}] call FUNC(addEventhandler);
+
+// Load settings on the server and broadcast them
 if (isServer) then {
     call FUNC(loadSettingsOnServer);
+    // Raise a local event for other modules to listen too
+    ["SettingsInitialized", []] call FUNC(localEvent);
```
Comment (Glowbal) [inline addons/common/XEH_preInit.sqf]: This is not true in case the server also is a client. It still has to set the client settings to correct values.

Comment (esteldunedain) [inline addons/common/XEH_preInit.sqf]: Yes, this event should be "ServerSettingsReceived", which also explains the comment above.

**Comment by esteldunedain** [comment]
Making a note to change this: https://github.com/acemod/ACE3/pull/1209/files#diff-c3ae8678638e2f52ea3ce4ea81ef3ca0R201

**Pull request merged**